api/src/main/java/jakarta/data/repository/DataRepository.java

```diff
@@ -25,5 +27,27 @@
  */
 public interface DataRepository<T, K> {
 
-
+    /**
+     * <p>Returns the set of reserved keywords that are supported by the
+     * database to which the repository connects. Reserved keywords begin
+     * with a capital letter as they appear within a method name
+     * and as defined by the Jakarta Data specification in its list of
+     * {@link Repository Reserved Keywords}. This set also includes the
+     * Jakarta Data provider's vendor-specific reserved keywords (if any)
+     * that are supported by the database.
+     * This set does not include reserved method name prefixes,
+     * such as <code>find...By</code> and <code>deleteBy</code>.</p>
+     *
+     * <p>The following reserved keywords must be supported for all
+     * databases:</p>
+     *
+     * <ul>
+     * <li><code>And</code></li>
+     * <li>TODO specify the remainder of this list in a separate pull once the exact list is determined</li>
+     * </ul>
+     *
+     * @return reserved keywords for repository method names that are
+     *         supported by the database.
+     */
+    Set<String> supportedKeywords();
```

Comment:
When we talk about Repositories, we primarily put the leading reference on DDD and the repository pattern.

DDD, a repository, is an object that participates in the domain but abstracts storage and infrastructure details away.

When we put a method like this, I don't believe we're putting the tech details away.